llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/TaskExecutorService.java

```diff
@@ -176,6 +181,31 @@ public TaskExecutorService(int numExecutors, int waitQueueSize,
     Futures.addCallback(future, new WaitQueueWorkerCallback());
   }
 
+  /**
+   * Sets the TaskExecutorService capacity to the new values. Both the number of executors and the
+   * queue size should be smaller than that original values, so we do not mess up with the other
+   * settings. Setting smaller capacity will not cancel or reject already executing or queued tasks
+   * in itself.
+   * @param newNumExecutors The new number of executors
+   * @param newWaitQueueSize The new number of wait queue size
+   */
+  @Override
+  public synchronized void setCapacity(int newNumExecutors, int newWaitQueueSize) {
+    if (newNumExecutors > configuredMaxExecutors) {
```

Comment:
Should we also verify that the new values are >= zero?

Reply:
Added check and test